src/twisted/conch/ssh/keys.py

```diff
@@ -942,24 +942,15 @@ def __init__(self, keyObject):
         """
         self._keyObject = keyObject
 
-    def __eq__(self, other):
+    def __eq__(self, other: object) -> bool:
```

Comment:
The return type is specified as `bool`, but this function can also return `NotImplemented`.
Is it correct to specify `bool` as the return type?

Reply:
I think this is handled as a special case by mypy. In typeshed, which is probably the closest thing we have to an authority on this, `bool` is also used as the return type for `__eq__`.

Reply:
Yes you are correct.  In mypy, `__eq__()` is part of `BINARY_MAGIC_METHODS`: https://github.com/python/mypy/blob/23dba19098857a72e0ad2284b164e2fceaee44ac/mypy/sharedparse.py#L55

 which is checked here: https://github.com/python/mypy/blob/259e0cfb65b656ce0847be7c41a3a12f7c8b00a0/mypy/checker.py#L3201

and with a specific exception for `builtins.NotImplemented` here:  https://github.com/python/mypy/blob/259e0cfb65b656ce0847be7c41a3a12f7c8b00a0/mypy/checker.py#L5053